NoSQL/database_design.txt

```diff
@@ -0,0 +1,105 @@
+1) Collection users:
+
+Design of single entry for user:
+
+{
+	_id : <user_id>,
+
+	credentials : {
+		email : <email_id>,
+		password : <password>
+	},
+
+	profile_details : {
+		name : {
+			first_name : <first_name>,
+			last_name : <last_name>
+		},
+
+		date_of_birth : <date>,
+
+		profile_picture : <url>,
+
+		contact : <contact>,
+
+		place : <city-state>,
+
+		<other_profile_details>
+	},
+
+	roles : {
+		position : <user/admin/editor/superadmin>,
```

Comment:
- admin should have access to a subset of users
- editor can have access to a subset of posts.